crates/ra_lsp_server/src/main_loop/handlers.rs

```diff
@@ -537,13 +537,16 @@ pub fn handle_formatting(
     let output = rustfmt.wait_with_output()?;
     let captured_stdout = String::from_utf8(output.stdout)?;
     if !output.status.success() {
-        failure::bail!(
-            r#"rustfmt exited with:
+        return Err(LspError::new(
+            gen_lsp_server::ErrorCode::ParseError as i32,
```

Comment:
Ow, I think this is the error code we should **not** return.

IIRC, this error code is postulated by the JSON-RPC protocol itself, and it means that the actual JSON is malformed.